lint-diff.js

```diff
@@ -75,7 +75,12 @@ var process_lintfile = function(filename) {
                 type: message.severity > 1 ? 'error' : 'warning',
                 message: message.message + ' (' + message.ruleId + ')'
             });
-            output.messages.push(message.ruleId + message.severity + message.source + message.message);
+            if (message.message.includes('on line')) {
+                output.messages.push(message.ruleId + message.severity + message.messageId);
```

Comment:
I think this will still lead to 'false positives' in the output because the sort algorithm is not based on the same fields as the output (in the case where 'on line' is in the message then the sorting will be still based on the line number text even though it is not printed).

What information is encoded in the 'messageId' field? Could this be used in place of the 'message' field? or is it representing something other than the message?

Reply:
The 'messageId' field seems to work similar as 'ruleId' seen [here](https://eslint.org/docs/latest/integrate/nodejs-api#-lintmessage-type). It is the message name associated with the 'message' field. 

I am unsure on what we could use other than the 'message' field to ensure no 'false positives'. The 'messageId' field alone isn't enough to tell if a new error replaced an old one such as with undefined variables. I am also unsure if using the 'source' field would be reliable. It is being used the the current code but its 'undefined' because the 'source' field is not within the 'messages' array, it is within the LintResult object itself (`report[0].source`). I sent you an output file on slack that contains what the eslint output looks like so that you can see what is contained in a LintResult object.